autoscheduler/frontend/src/components/SchedulingPage/Schedule/Schedule.tsx

```diff
@@ -231,46 +232,52 @@ const Schedule: React.FC<RouteComponentProps> = () => {
     </div>
   ));
 
-  // build each day based on schedule
-  function getMeetingsForDay(day: number): Meeting[] {
-    // meetingDays = MTWRFSU
-    // day = MTWRF
-    return schedule.filter((meeting) => meeting.meetingDays[day]);
-  }
-  function getAvailabilityForDay(day: number): Availability[] {
-    return availabilityList.filter((avl) => avl.dayOfWeek === day);
-  }
-  function renderMeeting(meeting: Meeting): JSX.Element {
-    const colors = ['#500000', '#733333', '#966666', '#b99999', '#dccccc'];
-    return (
-      <MeetingCard
-        meeting={meeting}
-        bgColor={colors[uniqueSections.indexOf(meeting.section.id)]}
-        key={meeting.id}
-        firstHour={FIRST_HOUR}
-        lastHour={LAST_HOUR}
-      />
+  // let's see if useMemo reduces our render time
+  const meetingsForDays = React.useMemo(() => {
+    const uniqueSections = [...new Set(schedule.map((mtg: Meeting) => mtg.section.id))];
+    // build each day based on schedule
+    function getMeetingsForDay(day: number): Meeting[] {
+      // meetingDays = UMTWRFS
+      // day = MTWRF
+      return schedule.filter((meeting) => meeting.meetingDays[day]);
+    }
+    function renderMeeting(meeting: Meeting): JSX.Element {
+      const colors = ['#500000', '#733333', '#966666', '#b99999', '#dccccc',
+        '#871b1e', '#9a1d26', '#c2777d', '#9f494b', '#b76778'];
+      return (
+        <MeetingCard
+          meeting={meeting}
+          bgColor={colors[uniqueSections.indexOf(meeting.section.id) % colors.length]}
+          key={meeting.id}
+          firstHour={FIRST_HOUR}
+          lastHour={LAST_HOUR}
+        />
+      );
+    }
+    return [0, 1, 2, 3, 4].map(
```

Comment:
Done for lines 257 and 277, but on line 16, that array is used somewhat differently, so I have left it as is for now. Line 16 uses an array of the one-letter keys of the days rather than an array of the numerical keys of the days. This would complicate the usage of the `DayOfWeek` enum, which has only numerical values.